### DISCUSSION TITLE
How to render highways?

### DISCUSSION THREAD

**Discussion by AtelierCartographique** [domlysz/BlenderGIS#709] [Q&A] [answered]
Highways are downloaded as a mesh, but I cannot find a way to actually make each road segment into a volume to which I could apply a material to render. What is the proper workflow to render road networks downloaded within the GET OSM menu?
Thank you so much!

**Comment by atseewal** [comment, marked as answer]
Geometry nodes might be a good way to accomplish this. If you select your highways and enter the geometry nodes tab, you can build a node tree that will take the OSM road networks and allow you to texture them with a material. Getting some inspiration from [this ](https://blender.stackexchange.com/questions/249310/geometry-node-curve-to-mesh-is-not-working)stack exchange post, here is the geometry node setup I came up with (I'm not a geometry nodes expert by any means):

1. Convert the road network mesh to curves
2. Convert the network curves back to mesh
    a. Use a Quadrilateral node set to rectangle to control the shape of the profile curve
    b. Adjust the Quadrilateral node size settings to make the roads wider and flatter
3. Use a Set Material node to set the material of the roads

The node tree:

![image](https://user-images.githubusercontent.com/39170343/192389827-a40ccc3f-5abb-49b7-8e06-7dc3770f40ae.png)

The output of this setup:

![image](https://user-images.githubusercontent.com/39170343/192389895-ade77562-2934-445c-ac49-5581b575b129.png)

**Reply by AtelierCartographique** [reply]
Thank you so much Atseewal, it works perfectly! :)
